### DISCUSSION TITLE
Subquery splat in SELECT statement

### DISCUSSION THREAD

**Discussion by ghost** [duckdb/duckdb#5692] [Ideas]
In bigquery, `select 'record 1' as record_name, (select as struct 1 as x, 2 as y).*` yields 

<img width="509" alt="Screen Shot 2022-12-13 at 6 26 47 PM" src="https://user-images.githubusercontent.com/6692076/207467177-69948a53-f09d-4c79-bd55-90e97b87cbd2.png">

In DuckDB, expanding splat to a subquery on select statement is not supported.

<img width="825" alt="Screen Shot 2022-12-13 at 6 28 14 PM" src="https://user-images.githubusercontent.com/6692076/207467363-3b6dd46f-6441-43e2-bd00-24f08f79f572.png">

Could this be supported? It's a convenient way to fetching a record and expanding it into the columns.

**Comment by Alex-Monahan** [comment]
Howdy! It may be that I am reading too much into this specific minimal example, but could this be written as this instead?
```sql
select 'record 1' as record_name, subquery.*
From (select 1 as x, 2 as y) subquery
```

**Reply by ghost** [reply]
yes, certainly!

I think the translation gets complex as the select statement grows

```sql
select
  'record 1' as record_name,
  (select as struct a as a1, b as b1 from t where var = 1).*,
  (select as struct a as a2, b as b2 from t where var = 2).*,
  (select as struct a as a3, b as b3 from t where var = 3).*;
```

these are automatically generated by code, so there's a dozen of them.

**Comment by l1t1** [comment]
your example is too simple to understand.  pls give more data
